gandalf.js

```diff
@@ -1,127 +1,128 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.